tools/transform.cpp

```diff
@@ -538,8 +538,16 @@ static void calculateAndInitConstants(Transform &t) {
   uint64_t min_access_size = 16;
   bool does_mem_access = false;
   bool has_load = false;
+  // The number of bits needed to encode pointer attributes
+  bits_for_ptrattrs = 0;
 
   for (auto fn : { &t.src, &t.tgt }) {
+    for (auto &v : fn->getInputs()) {
+      auto i = dynamic_cast<const Input*>(&v);
+      if (i && i->hasAttribute(Input::NoCapture))
+        bits_for_ptrattrs = 1;
```

Comment:
this pattern makes it harder to add other attributes. You need a has_foo variables (at least locally) to then sum them up.